### ISSUE TITLE
Dockerize AzerothCore

### ISSUE THREAD

**Issue by FrancescoBorzi** [azerothcore/azerothcore-wotlk#1123]
It would be nice to have a Docker system for AzerothCore so one can simply start a (group of) docker container(s) which will automatically compile and run the core.

This would help many development activities since it will make easier and faster to spin up an AzerothCore server instance together with all its requirements (mysql, libs, etc..).

Ideally the dockerized container(s) should not fetch the sources from the internet, instead it should get them from the main file system in order to let the developer quickly change the sources and re-run the container(s) to test the changes (for example [mounting a host Directory in a Docker container](https://stackoverflow.com/questions/23439126/how-to-mount-a-host-directory-in-a-docker-container).

**Comment by BarbzYHOOL** [comment]
I don't use docker, but there are other ways to get a similar result. An installation script (done by yehonal), a .deb, an .appimage, I don't know.

But what would be the end goal besides easy installation?

**Comment by Viste** [comment]
i can do it(already do it on with sunwellcore just for fun, but docker is bad idea... we get more issues with docker when noobs cant do something similar 
and more cry about RAM

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL @Viste of course Docker (or whatever alternative) should be only one additional option to use AzerothCore, not the only/default one.

The advantage of Docker is that you run your app inside an isolated environment, for example inside a linux-based container (even if you use windows or mac os x), so it will not affect your main system. For example, instead of installing a mysql server in your main (host) system, you just run a docker container with it and that will make your life much easier (faster setup, possibility to run multiple instances on the same machine, etc...).

Basically it's the same as running a virtual machine (e.g. virtualbox) with your app running inside, with the difference that using Docker it will consume way less RAM and other resources.

**Comment by Viste** [comment]
@FrancescoBorzi Unfortunately using of docker in my work and home test suggests otherwise. only more ram usage

**Comment by FrancescoBorzi** [comment]
@Viste do you mean more RAM usage compared to running it into a whole virtual machine?

**Comment by FrancescoBorzi** [comment]
close with: https://github.com/azerothcore/azerothcore-wotlk/commit/7bff043a1ba493c769fa0159fdeaa7900dbd827b

**Comment by FrancescoBorzi** [comment]
@Viste about the RAM usage, check:

https://github.com/azerothcore/azerothcore-wotlk/tree/master/docker#memory-usage

**Comment by Viste** [comment]
yes but i mean when it runs for week or more

**Comment by FrancescoBorzi** [comment]
@Viste that shouldn't be a difference between docker or non-docker
